src/challenge/ChallengeEditor.tsx

```diff
@@ -275,6 +300,23 @@ class ChallengeEditor
     // saving the guide is easy
     this.props.bookStore.store.save(this.state.guideMd, this.props.guidePath);
 
+    // saving the files
+    let updatedFiles = new Map(
+      this.props.bookNode.additionalFiles?.map((file, index) => {
+        return [
+          file.filename,
+          this.fileEditorRefs.get(file.filename)?.getValue() ||
+            this.state.additionalFilesLoaded[file.filename],
+        ];
+      })
+    );
+    this.setState({ additionalFilesLoaded: Object.fromEntries(updatedFiles) });
```

Comment:
this is new: use the refs to get any potentially changed file values from the components.